RIOT-OS/RIOT
# dist/tools: check for for broken links in files

### Contribution description

This PR provides a small shell script that uses the `wget` command to check for broken links in given files.

The documentation in `boards`, `cpu`, `drivers`, ... contains dozens of broken links. It should be checked for such broken links from time to time. This PR provides a very small tool that can detect such broken links in files specified as parameters or in files returned as changed files from `dist/tools/ci/changed_files.sh`.

As the check in some cases takes a considerable amount of time, it does not seem appropriate to integrate it into the static tests.

### Testing procedure

Check an board documentation, for example:
For example
```
dist/tools/urlcheck/check.sh boards/arduino-mega2560/doc.txt 
Checking boards/arduino-mega2560/doc.txt ...
Invalid URL in boards/arduino-mega2560/doc.txt: http://userpage.zedat.fu-
Invalid URL in boards/arduino-mega2560/doc.txt: https://raw.
Invalid URL in boards/arduino-mega2560/doc.txt: http://userpage.zedat.fu-
Invalid URL in boards/arduino-mega2560/doc.txt: http://automation.binarysage.net/?p=1515
Invalid URL in boards/arduino-mega2560/doc.txt: http://eng.aten.eu/products/Mobility-&-USB/USB-Converters/USB-to-Serial-
```

### Issues/PRs references
